python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -148,6 +180,11 @@ def source_record_to_value_list(self, source_record):
             if sub_schema.getField().getTypeString() == 'Angle':
                 values.append(
                     source_record.get(sub_schema.getKey()).asDegrees())
+            elif obj_id is not None and sub_schema.getField().getName() == "diaObjectId":
+                # diaObjectId is a special case that will not be found in a
+                # difference catalog returned from the Butler. Hence we
+                # explicitly set it here.
+                values.append(int(obj_id))
```

Comment:
What if `source_record` does have a diaObjectId field (e.g., because you add a call to `source_record_to_value_list` somewhere else)?

Reply:
If the obj_id isn't set then the field diaObjectId will just be read and stored. If obj_id is set then this value will be written to the DB.

Reply:
Is that the desired behavior, that you always overwrite the field if possible?